docs/content/new-tutorial/building-an-asset-graph.mdx

```diff
@@ -0,0 +1,221 @@
+    stopwords = set(STOPWORDS)
+    stopwords.update(["Ask", "Show", "HN"])
+    titles_text = " ".join([str(item) for item in hackernews_topstories["title"]])
+    titles_cloud = WordCloud(stopwords=stopwords, background_color="white").generate(titles_text)
+
+    # Generate the word cloud image
+    plt.figure(figsize=(8, 8), facecolor=None)
+    plt.imshow(titles_cloud, interpolation="bilinear")
+    plt.axis("off")
+    plt.tight_layout(pad=0)
+
+    # Save the image to a buffer
+    buffer = BytesIO()
+    plt.savefig(buffer, format="png")
+    image_data = base64.b64encode(buffer.getvalue())
+
+    return image_data
+```
+
+Right now, materializing the asset will save the file to local storage. In a real-world scenario, you would save the file to a cloud storage bucket. In a later section, you'll learn how to tell Dagster where to store the data by using IO managers.
+
+### Metadata and previews
+
+As mentioned, the asset is materialized in storage. You can add metadata to help users explore the asset and understand it. For example, you can add a preview of the asset to the metadata.
+
+By modifying your code to match the snippet below, you will add metadata to your asset.
+
+```python
+from dagster import MetadataValue, Output, asset
+
+@asset
+def hackernews_topstories_word_cloud(context, hackernews_topstories):
+    stopwords = set(STOPWORDS)
+    stopwords.update(["Ask", "Show", "HN"])
+    titles_text = " ".join([str(item) for item in hackernews_topstories["title"]])
+    titles_cloud = WordCloud(stopwords=stopwords, background_color="white").generate(titles_text)
+
+    # Generate the word cloud image
+    plt.figure(figsize=(8, 8), facecolor=None)
+    plt.imshow(titles_cloud, interpolation="bilinear")
+    plt.axis("off")
+    plt.tight_layout(pad=0)
+
+    # Save the image to a buffer and embed the image into Markdown content for quick view
+    buffer = BytesIO()
+    plt.savefig(buffer, format="png")
+    image_data = base64.b64encode(buffer.getvalue())
+
+    md_content = f"![img](data:image/png;base64,{image_data.decode()})"
+
+    # Attach the Markdown content as metadata to the asset
+    return Output(value=image_data, metadata={
+        "plot": MetadataValue.md(md_content)
+    })
+```
+
+For more information about metadata, you can read the [metadata documentation](/concepts/metadata).
+
+--- 
+
+## Adding more metadata
+
+You just used asset metadata to add a preview of the visualization asset. Software-defined assets can be enriched with different types of metadata. Dagster has some specific types of metadata that can be set. However, as you've seen, you can freely add any value as metadata to an SDA.
+
+A couple of Dagster's specific metadata are defining the asset group that an SDA belongs to or the type of computation it is. In the snippet below, the `hackernews_topstory_ids` asset that you wrote earlier has these metadata defined.
+
+```python
+@asset(group_name="hackernews", compute_kind="HackerNews API")
```

Comment:
💭  I like that this demonstrates asset groups, but asset groups not being mentioned after this makes it feel incomplete to me. What's the point of putting assets into groups? What would that look like in Dagit?

I don't think we necessarily need to go into detail here about this topic, but perhaps we can point users to asset group docs as a jumping-off point.

❓ Should we include an updated screenshot of Dagit that shows how the asset graph has changed?